api/src/pcapi/connectors/api_allocine.py

```diff
@@ -77,10 +97,24 @@ def _parse_movie_showtimes(
             raise
 
         data = json_data
-        for err in exc.errors():
-            data = filter_invalid_edge(data, err["loc"])
 
-        return _parse_movie_showtimes(data, try_count + 1)
+        # warning: the reversed sort is very import here. Without it, two (bad)
+        # things could happen:
+        #   1. IndexError because the json data has lost edges and the error
+        #      location index if out of date or
+        #   2. a valid showtime might be ignored (same problem as above but
+        #      with not-out-of-bound index).
+        # Starting from the highest index means that each other index is still
+        # valid after the filter call: eg, indexes are 1, 3, 5; after the 5th
+        # edge has been removed, the 3d will still be at the 3d index, same for
+        # the first one. Which would not be the case without the reversed sort.
+        filtered_locs = [_get_error_loc(err) for err in exc.errors() if _has_valid_error_loc(err)]
+        locs = sorted([loc for loc in filtered_locs if loc is not None], reverse=True)
```

Comment:
Il me semble que le filtre `loc is not None` n'est pas nécessaire car c'est loc `None` ont été filtrées par `_has_valid_error_loc `
```suggestion
        locs = sorted([loc for loc in filtered_locs], reverse=True)
```